src/Data/HKD/Distributive.hs

```diff
@@ -372,38 +381,91 @@ flogFromLogarithm = \ (FLogarithm f) -> f faskDist
-flogToLogarithm = \ f -> FLogarithm (ftraceDist f)
+flogToLogarithm = \f -> FLogarithm (ftraceDist f)
 {-# inline flogToLogarithm #-}
 
-{-
-newtype Foo f a = Foo { runFoo :: f (Const a) }
+-------------------------------------------------------------------------------
+-- Lower
+-------------------------------------------------------------------------------
+-- probably belongs in HKD, but then the Distributive instance becomes an orphan
+
+-- also consider adding Raise :: (Type -> Type) -> (() -> Type) -> Type 
+-- in the other direction?
+
+-- | Forget higher-kindedness. 
+-- Generally, if @f@ is an @FThing@ then @'Lower' f@ is a @Thing@
+newtype Lower f a = Lower { runLower :: f (Const a) }
 
-instance FFunctor f => Functor (Foo f) where
-  fmap f = Foo #. ffmap (Const #. f .# getConst) .# runFoo
+instance FFunctor f => Functor (Lower f) where
+  fmap = \f -> Lower #. ffmap (Const #. f .# getConst) .# runLower
   {-# inline fmap #-}
 
-newtype DScatter w f = DScatter (w (Foo f))
+-- Needs Contravariant
+-- instance FContravariant f => Contravariant (Lower f) where
+
+instance FFoldable f => Foldable (Lower f) where
+  foldMap = \f -> ffoldMap (f .# getConst) .# runLower
+  {-# inline foldMap #-}
+
+instance FTraversable f => Traversable (Lower f) where
+  traverse = \f -> fmap Lower . ftraverse (fmap Const . f .# getConst) .# runLower
+  {-# inline traverse #-}
+
+-- Assumes FRepeat is FApplicative
+instance FRepeat f => Applicative (Lower f) where
+  (<*>) = \(Lower fab) (Lower fa) -> Lower (fzipWith coerce fab fa) 
+  {-# inline (<*>) #-}
+  pure = \a -> Lower $ frepeat (Const a)
+  {-# inline pure #-}
+
+newtype DScatter w f = DScatter { runDScatter :: w (Lower f) }
 
 instance FFunctor w => FFunctor (DScatter w) where
+  ffmap = \f -> DScatter #. ffmap (Lower #. f .# runLower) .# runDScatter
+  {-# inline ffmap #-}
 
-instance FDistributive f => Distributive (Foo f) where
-  type Log (Foo f) = Some (FLog f)
-  -- scatter :: (w Identity -> r) -> (g ~> Foo f) -> w g -> Foo f r
-  scatter k g h = Foo $ _ -- fscatter k' g' h'
-  -- k :: w Identity -> r
-  -- g :: g ~> Foo f
-  -- h :: w g
-  ----------------------------
-  -- f (Const a)
-  --
-  -- fdistrib :: w f -> (w % Element ~> r) f r
-  -- r ~ Const a
+instance FDistributive f => Distributive (Lower f) where
+  type Log (Lower f) = Some (FLog f)
+  scatter = \k g -> Lower . fscatter (Const #. k .  ffmap coerce .# runDScatter) id . DScatter . ffmap g
   {-# inline scatter #-}
-  index = \fa (Some lg) -> getConst (findex (runFoo fa) lg)
+  index = \fa (Some lg) -> getConst (findex (runLower fa) lg)
   {-# inline index #-}
-  tabulate = \f -> Foo $ ftabulate (Const #. f . Some)
+  tabulate = \f -> Lower $ ftabulate (Const #. f . Some)
   {-# inline tabulate #-}
 
+-- maybe this could be replaced with `Some FLogarithm`, either using Flexible instances,
+-- or if we had EqSome and OrdSome classes, with EqSome f => Eq (Some f), etc.
```

Comment:
Is https://hackage.haskell.org/package/some-1.0.2/docs/Data-GADT-Compare.html the `EqSome` you are looking for (`some` is dependency of `hkd`), or does it have different method?

Reply:
I was thinking `eqSome :: f a -> f b -> Bool`, with no proof of `a :~: b` when they're equal. The Eq and Ord instances for Some don't require such a proof, so the constraint in that package is a little stronger than necessary. Though maybe that would be okay anyway, depending on whether we can safely produce such a proof (or are willing to unsafely produce it).